File: flut/.flutter/lib/flutter/widgets/icon_data.dart
import 'package:flutter/material.dart';
import 'package:flut/flut/object.dart';
import 'package:flut/flut/runtime.dart';

class FlutIconData extends FlutValueObject {
  final IconData iconData;

  const FlutIconData(this.iconData) : super('IconData');

  @override
  Map<String, dynamic> flutEncode() {
    final result = flutBaseProps();
    result['codePoint'] = iconData.codePoint;
    result['fontFamily'] = iconData.fontFamily;
    result['fontPackage'] = iconData.fontPackage;
    result['matchTextDirection'] = iconData.matchTextDirection;
    result['fontFamilyFallback'] = iconData.fontFamilyFallback;
    return result;
  }

  static IconData? flutDecode(FlutRuntime runtime, Map<String, dynamic> data) {
    return IconData(
      runtime.unpackRequiredField<int>(data, 'codePoint'),
      fontFamily: runtime.unpackOptionalField<String>(data, 'fontFamily'),
      fontPackage: runtime.unpackOptionalField<String>(data, 'fontPackage'),
      matchTextDirection: runtime.unpackRequiredField<bool>(
        data,
        'matchTextDirection',
      ),
      fontFamilyFallback: runtime
          .unpackOptionalField<List<dynamic>>(data, 'fontFamilyFallback')
          ?.cast<String>(),
    );
  }
}

File: flut/flutter/widgets/icon_data.py
from typing import Optional, override
from flut._flut.engine import FlutValueObject, _flut_pack_value
from flut._flut.runtime import _flut_unpack_optional_field, _flut_unpack_required_field


class IconData(FlutValueObject):
    _flut_type = "IconData"

    def __init__(
        self,
        codePoint: int,
        *,
        fontFamily: Optional[str] = None,
        fontPackage: Optional[str] = None,
        matchTextDirection: bool = False,
        fontFamilyFallback: Optional[list[str]] = None,
    ):
        super().__init__()
        self.codePoint = codePoint
        self.fontFamily = fontFamily
        self.fontPackage = fontPackage
        self.matchTextDirection = matchTextDirection
        self.fontFamilyFallback = fontFamilyFallback

    @staticmethod
    def _flut_unpack(data: dict) -> "IconData":
        return IconData(
            codePoint=_flut_unpack_required_field(data, "codePoint"),
            fontFamily=_flut_unpack_optional_field(data, "fontFamily"),
            fontPackage=_flut_unpack_optional_field(data, "fontPackage"),
            matchTextDirection=_flut_unpack_required_field(data, "matchTextDirection"),
            fontFamilyFallback=_flut_unpack_optional_field(data, "fontFamilyFallback"),
        )

    @override
    def _flut_pack(self) -> dict:
        result = self._flut_base_props()
        result["codePoint"] = _flut_pack_value(self.codePoint)
        result["matchTextDirection"] = _flut_pack_value(self.matchTextDirection)
        if self.fontFamily is not None:
            result["fontFamily"] = _flut_pack_value(self.fontFamily)
        if self.fontPackage is not None:
            result["fontPackage"] = _flut_pack_value(self.fontPackage)
        if self.fontFamilyFallback is not None:
            result["fontFamilyFallback"] = _flut_pack_value(self.fontFamilyFallback)
        return result
